script.js

```diff
@@ -1,4 +1,341 @@
+    cardDeck[randomIndex] = currentCard;
+    // increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // return the shuffled deck
+  return cardDeck;
+};
+
+// a function that deals two initial cards each to computer and player
+var initialDealRound = function () {
+  cardDeck = makeDeck();
+  shuffledDeck = shuffleCards(cardDeck);
+
+  // one card dealt each is first dealt to computer and player, then this repeats. so in total two cards each dealt to computer and player
+  // push 1st card from shuffled deck to computer's hand
+  computerHand.push(shuffledDeck.pop());
+  // push 1st card from shuffled deck to player's hand
+  playerHand.push(shuffledDeck.pop());
+  // push 2nd card from shuffled deck to computer's hand
+  computerHand.push(shuffledDeck.pop());
+  // push 2nd card from shuffled deck to player's hand
+  playerHand.push(shuffledDeck.pop());
+
+  // add up card scores to give computerTotalScore and playerTotalScore
+  // computerTotalScore = countHandScore(computerHand);
+  // playerTotalScore = countHandScore(playerHand);
+  computerTotalScore = computerHand[0].rank + computerHand[1].rank;
+  playerTotalScore = playerHand[0].rank + playerHand[1].rank;
+  console.log(`Computer's current total score: ${computerTotalScore}`);
+  console.log(`Player's current total score: ${playerTotalScore}`);
+
+  // output a message to display the four cards dealt so far
+  computerCardsMessage = `Computer's hand: <br>${computerHand[0].name} of ${computerHand[0].suit} <br>${computerHand[1].name} of ${computerHand[1].suit}`;
+  playerCardsMessage = `Player's hand: <br>${playerHand[0].name} of ${playerHand[0].suit} <br>${playerHand[1].name} of ${playerHand[1].suit}`;
+  // final message combines computerCardsMessage and playerCardsMessage to show all cards
+  message = `${computerCardsMessage} <br><br> ${playerCardsMessage}`;
+
+  // stores message from determineInitialWinner function
+  initialMessage = determineInitialWinner();
+  // concentenates the messages. 'initialDealRoundMessage' message will either: (1) say player wins Black Jack, or (2) message that tells user to enter 'hit' or 'stand'
+  initialDealRoundMessage = initialMessage + message;
+  return initialDealRoundMessage;
+};
+
+// a function that determines if player wins Black Jack after being dealt the first two cards. if player didn't win, program outputs message asking player to enter 'hit' or 'stand'.
+var determineInitialWinner = function () {
+  if (
+    ((playerHand[0].name == "ace" && playerHand[1].name == "ace") ||
+      (playerHand[0].name == "ace" && playerHand[1].rank == 10) ||
+      (playerHand[0].rank == 10 && playerHand[1].name == "ace")) &&
+    !(
+      (computerHand[0].name == "ace" && computerHand[1].name == "ace") ||
+      (computerHand[0].name == "ace" && computerHand[1].rank == 10) ||
+      (computerHand[0].rank == 10 && computerHand[1].name == "ace")
+    )
+  ) {
+    return "Player has Black Jack! You got 21. You win the game. <br><br>";
```

Comment:
(playerHand[0].name == "ace" && playerHand[1].name == "ace") value of the card will be 12 ( 11 + 1) or 2 ( 1+1) therefore having 2 ace card should not be blackjack player win.